processor/transformprocessor/README.md

```diff
@@ -230,6 +231,23 @@ Examples:
 
 - `convert_summary_sum_val_to_sum("cumulative", false)`
 
+## convert_gauge_to_histogram
+
+`convert_gauge_to_histogram(aggregation_temporality, bucket_count, lower_bound, upper_bound )`
+
+The `convert_gauge_to_histogram` function creates a new Histogram metric from the datapoints in a gauge.
+
+`aggregation_temporality` is a string (`"cumulative"` or `"delta"`) representing the desired aggregation temporality of the new metric. `bucket_count` is a integer representing the amount of buckets that will be created for the histogram. `lower_bound` is a float representing an explicit lower bound to create standardized bucket bounds. `upper_bound` is a float representing an explicit upper bound to create standardized bucket bounds. If `lower_bound` and `upper_bound` are equal, they will not be used, and instead the bucket bounds will be inferred based on the min, max, and bucket count.
+
+**NOTE:** This function may cause a metric to break semantics for [Histogram metrics](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#histogram). Use at your own risk.
```

Comment:
This seems like a very good reason not to do this.

I think we've had proposals before for a metric aggregation processor and that would be a good place for this functionality to like in a way that does not break the semantics of the metrics data model.  I would imagine this would only function correctly when multiple gauge data points are aggregated into a histogram based on dimension reduction.  Even then, that's probably not quite correct as it's not an aggregation of independent observations but of repeated observation of the same instrument.

@jmacd do you have any wisdom to impart regarding the differences between `histogram` and `gauge histogram`?

Reply:
I have not got a chance to review this PR or it's issue yet and I'm not familiar with OTEL histogram semantic conventions, but the processor itself does have precedence with unsound transformations: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/transformprocessor/README.md#warnings. 

The processor has some metric conversion functions already that were added in April/May after a lot of discussion on GitHub and at the SIG meetings. They all involve sums and gauges, though, which are simpler metric types in my mind.

Reply:
Extracting `sum` or `count` from a summary into a sum is one thing, though a gauge might be more appropriate.  This, however, is going to generate nonsense.  The bucket boundaries may be different every time it processes a set of data points.  The min/max/sum/count are set without regard for whether it's supposed to be cumulative or delta.  I'm all for giving users tools that can be used in ways that are correct and ways that are not, but it's not clear to me that a set of gauges can be sensibly converted to a histogram at all.

Reply:
> The bucket boundaries may be different every time it processes a set of data points.

Regarding this point, I've had trouble understanding the use-case/application of a delta aggregation temporality for Histogram instruments. The docs give an example of this exact combination, but the example only has one bucket, so the bounds are going to be uniform across all the observations. In my view this is not representative of Histogram usage in practice. Do you have insight into how delta Histograms would make sense? Are there any backend/storage systems that can make sense of Histograms that have bucket boundaries that may shift b/w observations? I'd love to understand this better.

The docs in question:

---
### Synchronous example: Delta aggregation temporality
Let’s imagine we export the metrics as [Histogram](https://opentelemetry.io/docs/reference/specification/metrics/data-model/#histogram), and to simplify the story we will only have one histogram bucket (-Inf, +Inf):

If we export the metrics using Delta Temporality:
```
(T0, T1]
attributes: {verb = GET, status = 200}, count: 2, min: 50 (ms), max: 100 (ms)
attributes: {verb = GET, status = 500}, count: 1, min: 1 (ms), max: 1 (ms)
(T1, T2]
nothing since we don’t have any Measurement received
(T2, T3]
attributes: {verb = GET, status = 500}, count: 2, min: 2 (ms), max: 5 (ms)
(T3, T4]
attributes: {verb = GET, status = 200}, count: 1, min: 100 (ms), max: 100 (ms)
(T4, T5]
attributes: {verb = GET, status = 200}, count: 3, min: 30 (ms), max: 100 (ms)
```
You can see that the SDK only needs to track what has happened after the latest collection/export cycle. For example, when the SDK started to process measurements in (T1, T2], it can completely forget about what has happened during (T0, T1].

Source: https://opentelemetry.io/docs/reference/specification/metrics/supplementary-guidelines/#synchronous-example-delta-aggregation-temporality

---

Reply:
Lets move the discussion to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18731